tools/pythonpkg/src/arrow_array_stream.cpp

```diff
@@ -37,18 +37,20 @@ unique_ptr<ArrowArrayStreamWrapper> PythonTableArrowArrayStreamFactory::Produce(
 	PythonTableArrowArrayStreamFactory *factory = (PythonTableArrowArrayStreamFactory *)factory_ptr;
 	D_ASSERT(factory->arrow_object);
 	py::handle arrow_obj_handle(factory->arrow_object);
+	auto scanner_class = py::module::import("pyarrow.dataset").attr("Scanner");
```

Comment:
Can this be a GetArrowType method that returns an enum, and turn these into a switch case?

I think an enum here would be nice  since it could also be used on other APIs that quack arrow

Reply:
Applied the feedback :) Could you have a look to see if this is what you had in mind?